src/model/Patient.ts

```diff
@@ -118,43 +166,174 @@ export default class Patient implements IPatient {
   get fullNameDisplay(): string {
     let names: Array<string> = [];
     if (this.firstName && this.firstName.length > 0) names.push(this.firstName);
-    if (this.lastName != null && this.lastName.length > 0) names.push(this.lastName);
+    if (this.lastName != null && this.lastName.length > 0)
+      names.push(this.lastName);
     return names.join(" ");
   }
 
+  get addressText(): string {
+    if (!this.address || !this.address.length) return "";
+    return this.address.find((a) => a.use === "home")?.text;
+  }
+
+  set addressText(text: string) {
+    if (this.address == null) this.address = [];
+    let address: IAddress = this.address.find((a) => a.use === "home");
+    if (address == null) {
+      address = { use: "home" } as IAddress;
+      this.address.push(address);
+    }
+    address.text = text;
+  }
+
   get homeZip(): string {
     return this.address != null && this.address.length > 0
-        ? this.address.find((a) =>
-            a.use === "home")?.postalCode : null;
+      ? this.address.find((a) => a.use === "home")?.postalCode
+      : null;
   }
 
   set homeZip(text: string) {
     if (this.address == null) this.address = [];
-    let address: IAddress =
-        this.address.find((a) => a.use === "home");
+    let address: IAddress = this.address.find((a) => a.use === "home");
     if (address == null) {
-      address = {use: "home"} as IAddress;
+      address = { use: "home" } as IAddress;
       this.address.push(address);
     }
     address.postalCode = text;
   }
 
   get secondZip(): string {
     return this.address != null && this.address.length > 0
-        ? this.address.find((a) =>
-            a.use !== "home")?.postalCode : null;
+      ? this.address.find((a) => a.use !== "home")?.postalCode
+      : null;
   }
 
   set secondZip(text: string) {
     if (this.address == null) this.address = [];
-    let address: IAddress =
-        this.address.find((a) => a.use !== "home");
+    let address: IAddress = this.address.find((a) => a.use !== "home");
     if (address == null) {
       address = {} as IAddress;
       this.address.push(address);
     }
     address.postalCode = text;
   }
+
+  addContact(
```

Comment:
To disambiguate, can you please name this and related functions to `*EmergencyContact`, eg "addEmergencyContact"